test/popup.js

```diff
@@ -0,0 +1,40 @@
+const assert = require('assert');
+const { URL } = require('url');
+
+const popup = require('../popup').testing;
+const mobileToCanonicalHost = popup.mobileToCanonicalHost;
+const getAddToListPostBody = popup.getAddToListPostBody;
+const constructLoginUrl = popup.constructLoginUrl;
+const parseTitleFromUrl = popup.parseTitleFromUrl;
+const parseDefaultListId = popup.parseDefaultListId;
+
+describe('popup.js', () => {
+
+    it('converts mobile to canonical hostnames', () => {
```

Comment:
A tricky one you might want to check is m.wikisource.org (versus, say, ml.wikisource.org).